### PR TITLE
feat: update SDK to 1.22.0 and add CI version sync check

### PR DESCRIPTION
**Pull request by malhotra5**
- [ ] A human has tested these changes.

---

## Why

The SDK has been updated to version 1.22.0. Additionally, we need CI to detect when the automation project's SDK dependencies drift from this repository's expected SDK version.

## Summary

- Update `DEFAULT_AGENT_SERVER_VERSION` from `1.21.1` to `1.22.0` in `scripts/dev-safe.mjs`
- Update SDK version references in `AGENTS.md`
- Add `scripts/check-sdk-version-sync.mjs` to verify automation project SDK dependencies match this repo
- Add `.github/workflows/sdk-version-sync.yml` CI workflow with multiple triggers:
  - Path-filtered PR/push triggers for version-related file changes
  - `repository_dispatch` triggers (`sdk-version-check`, `sdk-release`) for external repos to notify when SDK deps change
  - `workflow_dispatch` with optional version override
  - Scheduled runs every 6 hours

## Issue Number

N/A

## How to Test

```bash
# Run the SDK version sync check locally
node scripts/check-sdk-version-sync.mjs

# With PyPI version display
node scripts/check-sdk-version-sync.mjs --check-pypi

# Test with env var override
EXPECTED_SDK_VERSION=1.21.1 node scripts/check-sdk-version-sync.mjs

# Run the test suite
npm test
```

Note: The check will currently fail because the automation project still uses SDK 1.21.1 - the automation project needs to be updated separately.

## Video/Screenshots

N/A - CI/script changes

## Type

- [ ] Bug fix
- [x] Feature
- [ ] Refactor
- [ ] Breaking change
- [ ] Docs / chore

## Notes

The SDK version sync check will detect when the automation project (OpenHands/automation) uses different SDK versions than expected. External repos can trigger this check via:

```bash
curl -X POST -H "Authorization: token $GITHUB_TOKEN" \
  https://api.github.com/repos/OpenHands/agent-canvas/dispatches \
  -d '{"event_type": "sdk-version-check"}'
```

---
_This PR was created by an AI agent (OpenHands) on behalf of the user._

### CODE REVIEW

File: scripts/check-sdk-version-sync.mjs [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+    return null;
+  }
+}
+
+/**
+ * Fetch the pyproject.toml from the automation repository
+ */
+async function fetchAutomationPyproject() {
+  // The automation project is at https://github.com/OpenHands/automation
+  // We fetch its pyproject.toml from the main branch
```
[the remark this answers is not in the document]
Comment (malhotra5) [reply scripts/check-sdk-version-sync.mjs]: Fixed in 1eaaac3. The script now fetches from PyPI (more stable than GitHub), and the package name is configurable via `AUTOMATION_PACKAGE_NAME` env var.

File: scripts/check-sdk-version-sync.mjs [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+ *   openhands-workspace = "^1.22.0"
+ */
+function parseSdkVersions(pyprojectContent) {
+  const versions = {};
+
+  for (const pkg of SDK_PACKAGES) {
+    // Match various dependency formats:
+    // "pkg>=X.Y.Z" or "pkg==X.Y.Z" or "pkg~=X.Y.Z" or pkg = "^X.Y.Z"
+    const patterns = [
+      // Standard PEP 508 format: "pkg>=1.22.0" or "pkg==1.22.0"
```
[the remark this answers is not in the document]
Comment (malhotra5) [reply scripts/check-sdk-version-sync.mjs]: Fixed in 1eaaac3. Now uses:
1. `normalizeVersion()` to pad versions like `1.22` → `1.22.0` and strip pre-release metadata
2. `versionsEqual()` for semantic comparison
3. Simpler regex that first matches package name with `startsWith()`, then extracts version
4. Added test coverage for various formats including parenthesized `(>=1.22.0)` style

File: scripts/check-sdk-version-sync.mjs [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+    );
+  }
+  return { version: match[1], source: "dev-safe.mjs" };
+}
+
+/**
+ * Fetch the latest version of a package from PyPI
+ */
+async function fetchPyPIVersion(packageName) {
+  const url = `https://pypi.org/pypi/${packageName}/json`;
```
[the remark this answers is not in the document]
Comment (malhotra5) [reply scripts/check-sdk-version-sync.mjs:171]: Fixed in 1eaaac3. Added retry logic with exponential backoff:
- 3 retries by default
- Exponential delay (1s, 2s, 3s)
- No retry on 404 (configuration issue)
- Retries on 5xx and network errors

File: scripts/check-sdk-version-sync.mjs [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+
+    // Check each SDK package
+    let hasErrors = false;
+    let foundAny = false;
+    const mismatches = [];
+
+    console.log("Checking automation project SDK dependencies:");
+    console.log("");
+
+    for (const pkg of SDK_PACKAGES) {
```
[the remark this answers is not in the document]
Comment (malhotra5) [reply scripts/check-sdk-version-sync.mjs:347]: Fixed in 1eaaac3. Added:
- `normalizeVersion()` - pads to 3 parts (e.g., `1.22` → `1.22.0`), strips pre-release/build metadata
- `versionsEqual()` - uses normalized versions for comparison

This handles cases like `1.22` vs `1.22.0` as semantically equal.

File: .github/workflows/sdk-version-sync.yml [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v6
+        with:
+          node-version: 22.12.0
+
+      - name: Log trigger info
+        run: |
+          echo "Triggered by: ${{ github.event_name }}"
+          if [ "${{ github.event_name }}" = "repository_dispatch" ]; then
```
[the remark this answers is not in the document]
Comment (malhotra5) [reply .github/workflows/sdk-version-sync.yml:63]: Fixed in 1eaaac3. The workflow now uses:
```yaml
EXPECTED_SDK_VERSION: \${{ inputs.sdk_version || github.event.client_payload.version || '' }}
```
This checks `workflow_dispatch` inputs first, then `repository_dispatch` payload, then falls back to reading from dev-safe.mjs.

File: scripts/check-sdk-version-sync.mjs [hunk 6]
```diff
@@ -0,0 +1,307 @@
+#!/usr/bin/env node
```
[the remark this answers is not in the document]
Comment (malhotra5) [reply scripts/check-sdk-version-sync.mjs:1]: Fixed in 1eaaac3. Added `__tests__/scripts/check-sdk-version-sync.test.ts` with 19 tests covering:
- `normalizeVersion()` edge cases (padding, pre-release, build metadata)
- `versionsEqual()` semantic comparison
- `parseSdkVersionsFromRequiresDist()` with various PEP 508 formats
- SDK_PACKAGES constant validation
